UnitySDK/Assets/ML-Agents/Scripts/InferenceBrain/BarracudaModelParamLoader.cs

```diff
@@ -61,6 +63,26 @@ public static IReadOnlyList<TensorProxy> GetInputTensors(Model model)
             return tensors;
         }
 
+        public static int GetNumVisualInputs(Model model)
+        {
+            var count = 0;
+            if (model == null)
+                return count;
+
+            foreach (var input in model.inputs)
+            {
+                if (input.shape.Length == 4)
+                {
+                    if (input.name.StartsWith(TensorNames.VisualObservationPlaceholderPrefix))
```

Comment:
Magic strings... But I don't see an alternative without changing the way the trainer code works... @ervteng any refactor of the trainers (related to model inputs) planned?

Reply:
Yeah, I don't love it either. There are several other places in the (resurrected) barracuda code that check for this too. Trainer refactor is probably not sufficient, will likely need some tf2bc.py changes too.

Reply:
I wasn't planning on changing how the trainer detects these... but that was because I was assuming it wasn't going to change on the C# side :P would the alternative be having a "type" for the observation that's an enum or similar?

I think the tricky thing about the tf2bc part is that it uses the TensorFlow tensor names to determine what to put as input and output, and those are limited to being strings.